### PR TITLE
[Feature|Fix] Update Note flow

### PR DESCRIPTION
**Pull request by OleinikovArtem**
- Update notes.find_notes
- Add notes.find_notes_by_title
- Add notes.find_notes_by_tags
- Fix notes.edit_note
- Fix the error when we enter an empty string as cmd

### CODE REVIEW

**Review by thebedcoder — APPROVED** [review]

File: src/notes.py [hunk 1]
```diff
@@ -52,35 +52,55 @@ def add_note(self, note: Note):
...
 
-        self.data[current_note.get('title').value] = current_note
-        return
+        if new_title and old_title != new_title.lower():
+            del self.data[old_title]
+            self.data[new_title.lower()] = current_note
+        else:
+            self.data[old_title] = current_note
 
     def find_notes(self, key_words: list) -> list:
```
Comment (thebedcoder) [inline src/notes.py:78]: maybe we can also add search by content (description)?

File: src/notes.py [hunk 2]
```diff
@@ -52,35 +52,55 @@ def add_note(self, note: Note):
...
-        title = ' '.join(str(word) for word in key_words).lower()
+        notes_by_title = self.__find_notes_by_title(key_words)
+        notes_by_tags = self.__find_notes_by_tags(key_words)
 
+        return notes_by_title | notes_by_tags
+
+    def __find_notes_by_title(self, key_words: list):
+        title = ' '.join(str(word) for word in key_words).lower()
         result = dict()
```
Comment (thebedcoder) [inline src/notes.py:86]: why do we need dictionary here?

Comment (OleinikovArtem) [reply src/notes.py:86]: We have a dict format for list of notes, so I think using it in the same format when we use search is a good idea

File: src/notes.py (hunk 1 above)
Comment (OleinikovArtem) [reply src/notes.py:78]: We can, but it's not in the scope of the ticket. If I have time today, I will do it with another PR.

**Pull request merged**